src/data-sources/salesforce-d2c/SalesforceD2CSettings.tsx

```diff
@@ -2,44 +2,44 @@ import { TextControl } from '@wordpress/components';
 import { useMemo } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
-import { DataSourceForm } from '../components/DataSourceForm';
+import { DataSourceForm } from '@/data-sources/components/DataSourceForm';
 import PasswordInputControl from '@/data-sources/components/PasswordInputControl';
 import { useDataSources } from '@/data-sources/hooks/useDataSources';
 import {
+	SalesforceD2CConfig,
+	SalesforceD2CServiceConfig,
 	SettingsComponentProps,
-	SalesforceB2CConfig,
-	SalesforceB2CServiceConfig,
 } from '@/data-sources/types';
 import { useForm } from '@/hooks/useForm';
-import SalesforceCommerceB2CIcon from '@/settings/icons/SalesforceCommerceB2CIcon';
+import SalesforceCommerceD2CIcon from '@/settings/icons/SalesforceCommerceD2CIcon';
 
 const SERVICE_CONFIG_VERSION = 1;
 
-export const SalesforceB2CSettings = ( {
+export const SalesforceD2CSettings = ( {
 	mode,
 	uuid,
 	config,
-}: SettingsComponentProps< SalesforceB2CConfig > ) => {
-	const { onSave } = useDataSources< SalesforceB2CConfig >( false );
+}: SettingsComponentProps< SalesforceD2CConfig > ) => {
+	const { onSave } = useDataSources< SalesforceD2CConfig >( false );
 
-	const { state, handleOnChange, validState } = useForm< SalesforceB2CServiceConfig >( {
+	const { state, handleOnChange, validState } = useForm< SalesforceD2CServiceConfig >( {
 		initialValues: config?.service_config ?? {
 			__version: SERVICE_CONFIG_VERSION,
 			enable_blocks: true,
 		},
 	} );
 
 	const shouldAllowSubmit = useMemo( () => {
-		return state.shortcode && state.organization_id && state.client_id && state.client_secret;
-	}, [ state.shortcode, state.organization_id, state.client_id, state.client_secret ] );
+		return state.instance_url && state.store_id && state.client_id && state.client_secret;
+	}, [ state.instance_url, state.store_id, state.client_id, state.client_secret ] );
```

Comment:
Useless memo, there is no reason to memoize a simple boolean check. Just compute it. (I know this is existing code, this is a pet peeve of mine.)

```
const shouldAllowSubmit = state.instance_url && state.store_id && state.client_id && state.client_secret;
```

Reply:
Always better to follow the campground rule for existing code, so happy to change this up.